packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,232 @@
+                proc,
+                options.errorMatcherName
+            );
+
+            // when underlying process exits, notify tasks listeners
+            const toDispose = new DisposableCollection();
+
+            toDispose.push(
+                proc.onExit(event => {
+                    this.fireTaskExitedEvent({
+                        'taskId': task.id,
+                        'code': event.code,
+                        'signal': event.signal
+                    });
+                })
+            );
+
+            this.taskToDispose.set(task.id, toDispose);
+
+            return Promise.resolve(
+                {
+                    taskId: task.id,
+                    osProcessId: proc.pid,
+                    terminalId: (options.processType === 'terminal') ? proc.id : undefined,
+                    processId: proc.id
+                }
+            );
+
+        } catch (error) {
+            this.logger.error(`Error occured while creating task: ${error}`);
+            return Promise.reject(new Error(error));
+        }
+    }
+
+    protected fireTaskExitedEvent(event: ITaskExitedEvent) {
+        this.logger.debug(log => log(`task has exited:`, event));
+        if (this.client) {
+            this.client.onTaskExit(event);
+        }
+    }
+
+    public tasksListChanged(tasks: string[]) {
+        this.fireTaskChangedEvent({ 'tasks': tasks });
+    }
+
+    protected fireTaskChangedEvent(event: ITasksChangedEvent) {
+        this.logger.debug(log => log(`Configured tasks changed:`, event));
+        if (this.client) {
+            this.client.onTasksConfigChanged(event);
+        }
+    }
+
+    kill(id: number): Promise<void> {
+        const taskToKill = this.taskManager.get(id);
+        if (taskToKill !== undefined) {
+            taskToKill.kill();
+        }
+        return Promise.resolve();
+    }
+
+    /** Set the client we send the notifications-to */
+    setClient(client: ITaskClient | undefined) {
+        this.client = client;
+    }
+
+    // Check if task command can be found, if so return its resolved
+    // absolute path, else returns undefined
+    private findCommand(command: string, cwd: string): string | undefined {
+        const systemPath = process.env.PATH;
+
+        if (path.isAbsolute(command) && fs.existsSync(command)) {
+            return command;
+        } else if (path.isAbsolute(command)) {
+            const uri = new URI(cwd).resolve(command);
+            const resolved_command = FileUri.fsPath(uri);
+            if (fs.existsSync(resolved_command)) {
+                return resolved_command;
+            }
+        } else {
+            // look for command relative to cwd
+            const resolved_command = FileUri.fsPath(new URI(cwd).resolve(command));
+            if (fs.existsSync(resolved_command)) {
+                return resolved_command;
+            }
+
+            const separator = isWindows ? ";" : ":";
+
+            // just a command without path?
```

Comment:
I think it would just be simpler to try it out.

Reply:
True. The reason for findCommand() is try to figure-out if the command/script, we're about to try executing, can be found. We could just skip ahead and try to execute it; if it's then found and executed successfully, then no problem. But if it's not found or not successful, it can be difficult to distinguish what happened: 
 - the command was found and run, but exited with an error status (rc !=0)
 - the command was not found (rc != 0)

So, in findCommand() we try to determine if the command/script seems to exist, so that we can more correctly assume that any error is a result of its execution, not of it not having been found.

Reply:
OK I wish RawProcess/TerminalProcess would take care of that, I know I can know it for RawProcess but I don't think it's possible now for the Terminal.

So OK for now and we'll try to improve that upstream